azure-resource-optimization-toolkit/nestedtemplates/Automation.json

```diff
@@ -0,0 +1,929 @@
+        "name": "AutoSnooze_CreateAlert_Child",
+        "version": "1.0.0.0",
+        "description": "Runbook to create or disable alert for AutoSnooze",
+        "type": "PowerShell",
+        "Id": "",
+        "scriptUri": "[concat(parameters('_artifactsLocation'),'/scripts/Snooze/AutoSnooze_CreateAlert_Child.ps1')]"
+      },
+      "AutoSnooze_Disable": {
+        "name": "AutoSnooze_Disable",
+        "version": "1.0.0.0",
+        "description": "Disable the AutoSnooze",
+        "type": "PowerShell",
+        "Id": "",
+        "scriptUri": "[concat(parameters('_artifactsLocation'),'/scripts/Snooze/AutoSnooze_Disable.ps1')]"
+      },
+      "DeleteResourceGroups_Parent": {
+        "name": "DeleteResourceGroups_Parent",
+        "version": "1.0.0.0",
+        "description": "This runbook will delete resource groups",
+        "type": "PowerShell",
+        "Id": "",
+        "scriptUri": "[concat(parameters('_artifactsLocation'),'/scripts/DeleteRG/DeleteResourceGroups_Parent.ps1')]"
+      },
+      "DeleteResourceGroup_Child": {
+        "name": "DeleteResourceGroup_Child",
+        "version": "1.0.0.0",
+        "description": "This runbook will delete resource groups",
+        "type": "PowerShell",
+        "Id": "",
+        "scriptUri": "[concat(parameters('_artifactsLocation'),'/scripts/DeleteRG/DeleteResourceGroup_Child.ps1')]"
+      }
+    },
+      "Internal_AzureSubscriptionId": "Internal_AzureSubscriptionId",
+      "Internal_AROAutomationAccountName": "Internal_AROAutomationAccountName",
+      "Internal_AROResourceGroupName": "Internal_AROResourceGroupName",
+      "External_ExcludeVMNames": "External_ExcludeVMNames",
+      "External_ResourceGroupNames": "External_ResourceGroupNames",     
+      "External_AutoSnooze_Condition": "External_AutoSnooze_Condition",       
+      "External_AutoSnooze_Description": "External_AutoSnooze_Description",      
+      "External_AutoSnooze_MetricName": "External_AutoSnooze_MetricName",     
+      "External_AutoSnooze_Threshold": "External_AutoSnooze_Threshold",   
+      "External_AutoSnooze_TimeAggregationOperator": "External_AutoSnooze_TimeAggregationOperator",   
+      "External_AutoSnooze_TimeWindow": "External_AutoSnooze_TimeWindow",   
+      "azureCredentials": "AzureCredentials",
+      "AROToolkitVersion": "1.0.0.0",
+      "Internal_OMSWorkspaceId": "Internal_OMSWorkspaceId"
+      },
+      "resources": [
+      {
+      "apiVersion": "2015-10-31",
+      "location": "[parameters('AutomationRegion')]",
+      "name": "[parameters('AutomationAccountName')]",
+      "type": "Microsoft.Automation/automationAccounts",
+      "tags": {"AROToolkitVersion": "[variables('AROToolkitVersion')]"},
+      "comments": "Automation account",
+      "properties": {
+        "sku": {
+          "name": "Basic"
+        }
+      },
+        "resources": [
+         {
+            "name": "[variables('External_ExcludeVMNames')]",
+            "type": "variables",
```

Comment:
Is the Property Iteration section in this article an example of what you are talking about?  https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-create-multiple

Yeah there's a lot of duplication.  We could do the variables and runbooks I think.  The modules might be hard because they more unique depends on situations.  Let me know if I'm understanding your suggestion correctly and I can start up on the re-write.  This one might take a bit longer.